monai/transforms/spatial/array.py

```diff
@@ -1084,17 +1083,13 @@ def __call__(
         else:
             affine = self.affine
 
-        if isinstance(affine, np.ndarray):
-            affine = torch.as_tensor(np.ascontiguousarray(affine))
+        if self.device not in (None, torch.device("cpu"), "cpu"):
```

Comment:
Hi @rijobro ,

May I know why you check the device type here? I don't quite understand the logic.

Thanks in advance.